### PR TITLE
[12.0] ADD attachment_db_by_checksum

### PR DESCRIPTION
**Pull request by eLBati**
DB attachments saved by checksum

Allow to identify database attachments through their hash, avoiding duplicates

### CODE REVIEW

**Comment by eLBati** [comment]
Does anybody know what's happening to travis?

File: attachment_db_by_checksum/models/ir_attachment.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    _inherit = "ir.attachment"
+
+    @api.model
+    def _file_write(self, value, checksum):
+        location = self._storage()
+        if location != "hashed_db":
+            return super(Attachment, self)._file_write(value, checksum)
+        fname, _ = self._get_path(False, checksum)
+        att = self.env["ir.attachment.content"].search([
+            ("checksum", "=", fname)
```
Comment (simahawk) [inline attachment_db_by_checksum/models/ir_attachment.py]:
```suggestion
            ("checksum", "=", fname), limit=1
```
same for all similar search (which could be delegated to `_find_attachment_content`

File: attachment_db_by_checksum/models/ir_attachment.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @api.model
+    def _file_read(self, checksum, bin_size=False):
+        location = self._storage()
+        if location != "hashed_db":
+            return super(Attachment, self)._file_read(checksum, bin_size)
+        att = self.env["ir.attachment.content"].search([
+            ("checksum", "=", checksum)
+        ])
+        if not att:
+            _logger.info("File %s not found" % checksum)
```
Comment (simahawk) [inline attachment_db_by_checksum/models/ir_attachment.py]:
```suggestion
            _logger.debug("File %s not found" % checksum)
```

**Review by gurneyalex — APPROVED** [review]

**Review by simahawk — COMMENTED** [review]
try rebasing to fix the build

File: attachment_db_by_checksum/models/ir_attachment.py (hunk 1 above)
Comment (eLBati) [reply attachment_db_by_checksum/models/ir_attachment.py]: Done

File: attachment_db_by_checksum/models/ir_attachment.py (hunk 2 above)
Comment (eLBati) [reply attachment_db_by_checksum/models/ir_attachment.py]: Done

**Comment by eLBati** [comment]
@simahawk done thanks

**Pull request closed without merging**